pkg/publicapi/apimodels/error.go

```diff
@@ -67,31 +66,37 @@ func (e *APIError) Error() string {
 }
 
 // Parse HTTP Resposne to APIError
-func FromHttpResponse(resp *http.Response) (*APIError, error) {
-
+func GenerateAPIErrorFromHTTPResponse(resp *http.Response) *APIError {
 	if resp == nil {
-		return nil, errors.New("response is nil, cannot be unmarsheld to APIError")
+		return NewAPIError(0, "API call error, invalid response")
 	}
 
 	defer resp.Body.Close()
```

Comment:
I will preface this with: I am aware you are not the one that introduced this :smiley: . But the diff here caused be to notice a couple things.

We are closing the response body both inside this method and outside of it, e.g. in the file edit below on [line 88](https://github.com/bacalhau-project/bacalhau/pull/4565/files#diff-202b9471f43edeb5de262a1accf8cbe2be89e7af88bed8b70dd7d8b5cb0abaebR83). Probably best to avoid closing the response within this method, and leave it up to the callers/reduce unexpected side affects. Mind making that change here or in a follow on?